### PR TITLE
Support SingletonSymNode mul with coefficient

### PR DESCRIPTION
**Pull request by soulitzer**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #110529
* #110528
* #109171
* #110372
* #110371
* #110370
* __->__ #110369
* #110044

We want to be able to use SingletonSymNode to represent strides for Jagged layout tensor. The following is for 3D, but easily generalizable to higher dimensions.

Constraints:
- [B, x, D] (where x represents the "variably lengthed dim") can be strided in two ways [x, 1, sum(x)] and [dx, d, 1]. We need two different placeholder values depending on how the jagged tensor is strided.
- When doing operations we need the strides of output tensors to be expressable in terms of the strides and sizes of the inner tensors. Given [B, x, D] @ [D, D'], the output strides is [x * D', D', 1] rather than some opaque [x2, D', 1]. This constraint exists because if I'm tracing, I need a symint to represent the output stride. This symint needs to come from somewhere; I get it in several ways: (1) create a constant, (2) unbacked symint, (3) create a new input using a source, (4) output of an operation on an existing symint. It is clear that (4) is what we want here, which brings us to the design below.

Design:

Given the two constraints, the most straightforward way to implement this is actually to update SingletonSymNode to include some scalar factor, i.e. Morally, SingletonSymNode represents `factor * [s_0, s_1, …, s_n]` This enables us to symbolically compute strides from sizes.

### CODE REVIEW

File: torch/nested/_internal/nested_tensor.py [hunk 1]
```diff
@@ -9,12 +9,12 @@
 _tensor_id_registry = WeakTensorKeyDictionary()
 
 
-def get_tensor_id(tensor):
+def get_tensor_id(tensor, coeff):
```
Comment (ezyang) [inline torch/nested/_internal/nested_tensor.py:12]: kwarg me

**Review by ezyang — APPROVED** [review]
I didn't carefully audit to make sure coeffs were used everywhere they needed to be. kwarg-only coef would be nice.

File: torch/nested/_internal/nested_tensor.py (hunk 1 above)
Comment (soulitzer) [reply torch/nested/_internal/nested_tensor.py]: updated!

**Pull request closed without merging**